nautobot/core/apps/__init__.py

```diff
@@ -109,6 +114,77 @@ def register_menu_items(tab_list):
         )
 
 
+def register_homepage_panels(app_name, homepage_layout):
+    """
+    Register homepage panels using `homepage.py`.
+
+    Each app can now register a `homepage.py` file which holds objects defining the layout of the
+    home page. `HomePagePanel`, `HomePageGroup` and `HomePageItem` can be used to
+    define different parts of the layout.
+
+    These objects are converted into a dictionary to be stored inside of the Nautobot registry.
+    """
+    name, app_name = app_name.split(".")
+    template_path = f"{name}/{app_name}/templates/{app_name}/inc/"
+    registry_panels = registry["homepage_layout"]["panels"]
+    for panel in homepage_layout:
+        panel_perms = registry_panels[panel.name]["permissions"] if registry_panels.get(panel.name) else set()
+        if panel.permissions:
+            panel_perms |= set(panel.permissions)
+        panel.template_path = template_path
+        if isinstance(panel, HomePagePanel):
+            create_or_check_entry(registry_panels, panel, panel.name, f"{panel.name}")
+            registry_items = registry_panels[panel.name]["items"]
+
+            for item in panel.items:
+                if isinstance(item, HomePageItem):
+                    item.template_path = template_path
+                    create_or_check_entry(registry_items, item, item.name, f"{panel.name} -> {item.name}")
+                    panel_perms |= set(perms for perms in item.permissions)
+                elif isinstance(item, HomePageGroup):
+                    item.template_path = template_path
+                    create_or_check_entry(registry_items, item, item.name, f"{panel.name} -> {item.name}")
+                    for group_item in item.items:
+                        if isinstance(group_item, HomePageItem):
```

Comment:
Do we need an `else: raise TypeError` here, rather than silently ignoring anything that isn't a `HomePageItem`?